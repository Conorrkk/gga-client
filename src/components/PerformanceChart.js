import {
    Chart as ChartJS,
    BarElement,
    CategoryScale, // x axis
    LinearScale, // y axis
    Tooltip,
    Legend,
  } from "chart.js";
  
  import { Bar } from "react-chartjs-2";
  
  ChartJS.register(BarElement, CategoryScale, LinearScale, Tooltip, Legend);

function PerformanceChart({ data }) {
    if (!data) {
        return <div>Loading data...</div>;
      }
    
      if (data.length === 0) {
        return <div className="d-flex justify-content-center">Select a stat to view your players' performance</div>;
      }

      let stat = data[2].statName

    const performanceData = {
        labels: data.map((item) => item.name),
        datasets: [
          {
            label: `${stat}`,
            data: data.map((item) => item.amount),
            backgroundColor: "red",
            borderColor: "black",
            borderWidth: 1,
          },
        ],
      };
    
      const options = {
        plugins: {
          legend: true,
        },
        scales: {
          y: {
            min: 0,
            max: 16,
          },
        },
      };


    return (
        <div>
        <Bar
          style={{ width: "100%" }}
          data={performanceData}
          options={options}
        ></Bar>
      </div>
    )
}

export default PerformanceChart;