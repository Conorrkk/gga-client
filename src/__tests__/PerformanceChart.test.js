import React from 'react';
import renderer from 'react-test-renderer';
import PerformanceChart from '../components/PerformanceChart';

const dataMock = [
    {
        name: "John",
        amount: 8,
        statName: "Points from play"
    },
    {
        name: "Conor",
        amount: 8,
        statName: "Points from play"
    },
    {
        name: "Paul",
        amount: 8,
        statName: "Points from play"
    }
]

it('renders correctly', () => {
    const tree = renderer
    .create(<PerformanceChart data={dataMock}/>)
    .toJSON();

    expect(tree).toMatchSnapshot();
})