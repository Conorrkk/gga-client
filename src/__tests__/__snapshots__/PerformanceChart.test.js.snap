// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <canvas
    height={150}
    role="img"
    style={
      Object {
        "width": "100%",
      }
    }
    width={300}
  />
</div>
`;
